TestResultSummaryService/perf/BenchmarkMathCalculation.js

```diff
@@ -0,0 +1,144 @@
+class BenchmarkMathCalculation {
+
+    // Taken from Perffarm/perfsite/benchmarks.php
+    /*
+    * We need to calculate the standard deviation as part of the
+    * statistical tests for benchmarks so this support funciton
+    * is provided to ensure we are consistant.
+    */
+    static stddev( scores ) {
+
+        /* Make sure we are working with floating point
+        * numbers.
+        */
+        //There is also a minor check here that makes sure that
+        //we dont include NULL scores in the sd calculations.
+        //We wont need this when the annotation tool is being used
+        //as this should catch all 'bad' data.
+        //This is only partially correct anyway because there are no checks
+        //within the ci functions and so the data will still be wrong
+        //and include NULL data.
+        if (scores !== null) {
+            let bad_data = 0;
+            let scores_sum = 0;
+
+            scores.forEach((x) => {
+                if (x === null) {
+                    bad_data++
+                } else {
+                    scores_sum += x;
+                }
+            });
+
+            let count = scores.length - bad_data;
```

Comment:
Even though we'll be removing this `stddev` function in order to use the similar function from `mathjs`, we need to do a similar check in `BenchmarkParser.js` before using the standard library functions. We need to make sure that all values in the array are good (i.e. not null) before we call all the math functions such as `max`, `stddev` and others.